sip_assembly/routines.py

```diff
@@ -56,16 +56,23 @@ class BaseRoutine(object):
     """Base routine which contains main run method."""
 
     def run(self):
-        sip = SIP.objects.filter(process_status=self.start_status).first()
-        if sip:
-            try:
-                message = self.process_sip(sip)
-                sip.process_status = self.end_status
+        if not SIP.objects.filter(process_status=self.in_process_status).exists():
+            sip = SIP.objects.filter(process_status=self.start_status).first()
+            if sip:
+                sip.process_status = self.in_process_status
                 sip.save()
-            except Exception as e:
-                raise Exception(str(e), sip.bag_identifier)
+                try:
+                    message = self.process_sip(sip)
+                    sip.process_status = self.end_status
+                    sip.save()
+                except Exception as e:
+                    sip.process_status = self.start_status
+                    sip.save()
+                    raise Exception(str(e), sip.bag_identifier)
```

Comment:
I think how you did this in Pictor is better:

```suggestion
                try:
                    message = self.process_sip(sip)
                except Exception as e:
                    sip.process_status = self.start_status
                    sip.save()
                    raise Exception(str(e), sip.bag_identifier)
                 sip.process_status = self.end_status
                 sip.save()
```